From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Sun, 6 Oct 2024 15:57:13 -0500
Subject: [PATCH] Stateless dconf

We can make this work a bit more like gdm by compiling the dconf db ourselves and installing it to /usr in the build

---
 data/dconf/Makefile.am  | 5 +----
 data/dconf/profile/ibus | 2 +-
 2 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/data/dconf/Makefile.am b/data/dconf/Makefile.am
index b0601aa8..24b9afe2 100644
--- a/data/dconf/Makefile.am
+++ b/data/dconf/Makefile.am
@@ -26,12 +26,9 @@ gsettingsconvertdir = $(datadir)/GConf/gsettings
 dist_gsettingsconvert_DATA = ibus.convert
 @GSETTINGS_RULES@
 
-dconfprofiledir = $(sysconfdir)/dconf/profile
+dconfprofiledir = $(datadir)/dconf/profile
 dconfprofile_DATA = profile/ibus
 
-dconfdbdir = $(sysconfdir)/dconf/db/ibus.d
-dconfdb_DATA = 00-upstream-settings
-
 00-upstream-settings: $(srcdir)/make-dconf-override-db.sh | $(gsettings_SCHEMAS)
 	@$(MKDIR_P) db
 	$(AM_V_GEN) $(srcdir)/make-dconf-override-db.sh > $@ || \
diff --git a/data/dconf/profile/ibus b/data/dconf/profile/ibus
index aa8fe5f0..dfa55912 100644
--- a/data/dconf/profile/ibus
+++ b/data/dconf/profile/ibus
@@ -1,2 +1,2 @@
 user-db:user
-system-db:ibus
+file-db:/usr/share/ibus/ibus-dconf-defaults
